### ISSUE TITLE
Release Notes - PR release notes only for first dep in a monorepo

### ISSUE THREAD

**Issue by setchy** [renovatebot/renovate#10954]
**How are you running Renovate?**

- [x] WhiteSource Renovate hosted app on github.com
- [x] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [x] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [ ] Gitea
- [x] github.com
- [ ] GitHub Enterprise Server
- [ ] gitlab.com
- [ ] GitLab self-hosted

Renovate version: @latest (25.56.10 at time of post)

**Describe the bug**
We have a nodejs project that has dependencies on both
* `@apollo/federation` and 
* `@apollo/gateway`

It appears that when renovate-bot is producing the PR release notes is not handling both of the `CHANGELOG.md` files in the monorepo for each respective depdency
* https://github.com/apollographql/federation/blob/main/federation-js/CHANGELOG.md
* https://github.com/apollographql/federation/blob/main/gateway-js/CHANGELOG.md

Minimal reproduction details below
* Repository - https://github.com/setchy/renovate-bot-testbed-apollo
* Example PR - https://github.com/setchy/renovate-bot-testbed-apollo/pull/3

Only the `apollographql/federation` CHANGELOG.md notes are added, while the `apollographql/gateway` CHANGELOG.md is skipped entirely 
...

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [x] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

...

**Comment by setchy** [comment]
@rarkins - please let me know if you require further information to triage this

**Comment by rarkins** [comment]
I don't see any minimal reproduction repository (please see the linked document above). I believe it's related to the way we de-dupe packages from the same repo

**Comment by setchy** [comment]
@rarkins - the reproduction repository is this one - https://github.com/setchy/apollo-federation-poc/pull/17

**Comment by rarkins** [comment]
The presence of 13 dependencies disqualifies it as minimal

**Comment by setchy** [comment]
Got it.  Fixed
* Repository - https://github.com/setchy/renovate-bot-testbed-apollo
* Example PR - https://github.com/setchy/renovate-bot-testbed-apollo/pull/3

**Comment by rarkins** [comment]
I *think* it's this logic that's causing us to dedupe: https://github.com/renovatebot/renovate/blob/aa4fabe5997e07b2cb4f9f1b5eca8ed9810fc059/lib/workers/pr/index.ts#L317-L329

The further challenge is that we display release notes in PRs grouped by repo today, so that would need to change because monorepos can have different packages with different release notes.

**Comment by rarkins** [comment]
To progress from status:requirements to status:ready we'd need agreement on how the logic and presentation should work once we move from the assumption of source URLs being the "key" for each release notes.

**Comment by setchy** [comment]
Resolved with #12543
